### DISCUSSION TITLE
[RFC] Improve learnability of "Scrape with..." button (and supported scraper types)

### DISCUSSION THREAD

**Discussion by 7dJx1qP** [stashapp/stash#1905] [Ideas]
One of the most common questions that I see in the discord is "I installed a scraper but I don't see it in scrape with". The UI should do a better job allowing users to discover the answer to this for themselves so they don't have to come to the discord and get told to read the CommunityScrapers readme.

Ideas and areas for change:

- Rename "Scrape with..." to something more descriptive. Maybe something like "Metadata Scrape" to go along with the settings page listing them as "Scene Metadata" under supported types. I'm not sure this will help much though.
- A dropdown option in the "Scrape with..." list that's labeled "Missing scrapers?"
![image](https://user-images.githubusercontent.com/38586902/138960025-1b773c3d-04a2-4d44-a6a3-a6deb8f30b8c.png)
It could either:
  - Directly link to https://github.com/stashapp/CommunityScrapers#scrapers
  -  Display the same CommunityScrapers readme info in a popup dialog, similar to what the current help button does
  - Take the user to the Settings -> Scrapings page `/settings?tab=scraping` with the relevant list opened up (scene, gallery, performer, movie)
- Embed the CommunityScrapers readme info in the Settings -> Scraping page
- Update the Help dialog "Metadata Scraping" section with the CommunityScrapers readme info.
- Create an FAQ section, similar to the Help dialog or as part of the Help dialog, with answers to specific problems like "I don't see a scraper". This can be linked to in other parts of the app (like the dropdown option described above, or from the Settings -> Scrapings page)
- Add a more detailed breakdown of supported types to https://github.com/stashapp/CommunityScrapers/blob/master/SCRAPERS-LIST.md like the tables you can find in Settings -> Scraping
- Clarify how each supported type (Scene Metadata, Search by name, URL, etc) maps to what you will see in the stash UI.
  - Currently we just have this https://github.com/stashapp/stash/blob/develop/ui/v2.5/src/docs/en/Scraping.md#scraper-configuration-file-format which uses the technical scraper configuration property names like "sceneByFragment"

**Comment by philpw99** [comment]
Nah... I think "Scrape with..." is perfect. I don't even understand what the hell is "metadata". Sounds like some fancy words. Ain't it?
My other favorite name for this is "website data retriever", but it's too long. "Scrape" is just right.
For newbies who wants to know how to install scrapers and how to use them, use my Stash_Helper. Problem solved. 😄

**Comment by Belleyy** [comment]
I won't mind the "Missing scrapers?" in dropdown. It should have a CSS class to it so advanced user can remove it with custom CSS if they don't want to see it.
I think having the link to the repo could be better:
1. You have the readme.
2. If people don't know where to find scraper, now they have the page.

I think a FAQ section could also be useful and more general use. I am sure we can have tons of question answer and not only scraper related.

**Comment by WithoutPants** [comment]
I think the Scraping manual page needs to be separated into two pages - one for operation and one for writing scrapers. As it is it's way too much information and not newbie friendly. 

#623 will go a long way to eliminate _some_ of the common issues encountered by users installing scrapers for the first time.

I have toyed with the idea of tutorial overlays for new users. This would explain some of the less obvious features of the various pages - such as the URL scrape button, the filter bookmarks button and so on, with the option to turn them off once the user is accustomed.

Finally, I think we need to encourage the Q+A discussions of GitHub. Common questions like this can be posted, answered and upvoted, and it should become the _first_ place to check. The documentation should be amended to encourage this.
